erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -649,6 +650,12 @@ def get_gl_entries(self, warehouse_account):
 		gl_entries = super(StockEntry, self).get_gl_entries(warehouse_account)
 
 		total_basic_amount = sum([flt(t.basic_amount) for t in self.get("items") if t.t_warehouse])
+
+		if self.get("additional_costs") and not total_basic_amount:
+			#If additional costs table is populated and total basic amount is
+			#somehow 0, interrupt transaction.
+			frappe.throw(_("Total Basic Amount in Items Table cannot be 0"), TotalBasicAmountZeroError)
```

Comment:
Actually it can be zero, if "Allow zero valuation rate" is mentioned.

I think, in that case, we can "total_qty" for proportionately divide additional charges.

Reply:
In that case if total_basic_amount is zero , all the **basic_amount** fields are also 0, **(t.amount * d.basic_amount) / total_basic_amount** will always be zero (given that the denominator>0). No point in passing **total_qty** into a loop that is always setting the value as 0?

Reply:
done